_content/tour/ukr/composition-pollution.article

```diff
@@ -1,86 +1,86 @@
-Interface Pollution
-Interface pollution comes from the fact that people are designing software with interfaces instead of discovering them.
+Забруднення Інтерфейсами (Interface Pollution)
+Забруднення Інтерфейсами випливає з факту того, що люди створюють дизайн додатків за допомогою інтерфейсів замість того щоб відкривати його природно.
 
-* Interface Pollution
+* Забруднення Інтерфейсами
 
-- [[https://www.ardanlabs.com/training/individual-on-demand/ultimate-go-bundle/][Watch The Video]]
-- Need Financial Assistance, Use Our [[https://www.ardanlabs.com/scholarship/][Scholarship Form]]
+- [[https://www.ardanlabs.com/training/individual-on-demand/ultimate-go-bundle/][Дивитись відео]]
+- Якщо потрібна фінансова допомога, то використовуйте [[https://www.ardanlabs.com/scholarship/][форму для отримання стипендії]]
 
-Interface pollution comes from the fact that people are designing software with
-interfaces instead of discovering them.
+Забруднення інтерфейсами випливає з факту того, що люди створюють дизайн програмного забезпечення за допомогою інтерфейсів
+замість того, щоб відкривати його природно.
 
-** Code Review
+** Перегляд коду
 
-- *Example* *1:* Create Interface Pollution
-- *Example* *2:* Remove Interface Pollution
+- *Example* *1:* Створіть Забруднення Інтерфейсів
+- *Example* *2:* Приберіть Забруднення Інтерфейсів
 
 .play composition/pollution/example1.go
 .play composition/pollution/example2.go
 
-** Interface Pollution
+** Забруднення Інтерфейсами
 
-Interface pollution comes from the fact that people are designing software with
-interfaces instead of discovering them. You should design a concrete solution to the
-problem first. Then you can discover where the program needs to be polymorphic, if
-at all.
+Забруднення Інтерфейсами випливає з факту того, що люди проектують програмне забезпечення за допомогою інтерфейсів
+замість того, щоб відкривати його природно. Спочатку ви повинні створити
+конкретне рішення для проблеми. Потім ви можете зрозуміти де саме програма має буде поліморфною,
```

Comment:
```suggestion
конкретне рішення для проблеми. Потім ви можете зрозуміти, де саме програма має буде поліморфною,
```

https://onlinecorrector.com.ua/uk/%D0%BA%D0%BE%D0%BC%D0%B0-%D0%BC%D1%96%D0%B6-%D1%87%D0%B0%D1%81%D1%82%D0%B8%D0%BD%D0%B0%D0%BC%D0%B8-%D1%81%D0%BA%D0%BB%D0%B0%D0%B4%D0%BD%D0%BE%D0%B3%D0%BE-%D1%80%D0%B5%D1%87%D0%B5%D0%BD%D0%BD%D1%8F